### DISCUSSION TITLE
Shuffle Playlist URL (for one-button play start)

### DISCUSSION THREAD

**Discussion by dathbe** [navidrome/navidrome#2783] [Ideas]
I'd like to have a directly accessible URL to start playing (and, better, shuffling) a given playlist.  This would allow creating shortcuts so that a playlist could be started with one click.

**Comment by deluan** [comment]
That's a great idea, and I will try to implement it in the new UI. Thanks for the suggestion!

**Comment by dathbe** [comment]
I know it's not exactly a 1:1 comparison, but Play:Sub and Amperfy have implemented something similar.  So maybe this prevents entirely recreating the wheel.

Amperfy:
![2024-01-19 07-30-25 Darin_Phone 1450](https://github.com/navidrome/navidrome/assets/14321825/58cad814-1828-4b2c-bf06-de2ebf2ca6b7)
(Click "Settings", scroll to the bottom, and click "X-Callback-URL Documentation")

For whatever reason, I can't find the documentation on Play:Sub right not, but I recall reading both a Reddit post about it and a blog post from the developer.

I would hope it would be easier to implement directly through a web interface as opposed to the Subsonic API.  And I've had reliability isssues with Amperfy, so hopefully it's more reliable if we remove the Subsonic API middleman (though for all I know it's an iOS issue that will still remain).

**Reply by deluan** [reply]
Thanks for the pointers, but this is not part of the (Open)Subsonic API, it is a completely client side implementation.

In Navidrome it will be something implemented in the Web UI. As I mentioned, this is being considered for the new UI.

**Comment by bilogic** [comment]
Yes, I like a direct URL to play my favourite radio station without navigating through the menus, this also allows for better automation.
